### PR TITLE
builtin: add time function dayname()

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi there,

This PR adds builtin time function dayname(). Please note that the system variable ` lc_time_names` which controls the locales of output names is not supported.

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -112,6 +112,7 @@ import (
 	dateAdd		"DATE_ADD"
 	dateSub		"DATE_SUB"
 	day		"DAY"
+	dayname	"DAYNAME"
```
Comment (coocood) [inline parser/parser.y]: In `parser.y`, we use tab to align text.

File: parser/parser.y [hunk 2]
```diff
@@ -2182,6 +2183,10 @@ FunctionCallNonKeyword:
 	{
 		$$ =  &ast.FuncCallExpr{FnName: model.NewCIStr($1.(string)), Args: []ast.ExprNode{$3.(ast.ExprNode)}}
 	}
+|	"DAYNAME" '(' Expression ')'
+	{
+		$$ = &ast.FuncCallExpr{FnName: model.NewCIStr($1.(string)), Args: []ast.ExprNode{$3.(ast.ExprNode)}}
+	}
```
Comment (zimulala) [inline parser/parser.y]: add tests

File: expression/builtin/time.go [hunk 3]
```diff
@@ -162,6 +162,21 @@ func builtinNow(args []interface{}, ctx map[interface{}]interface{}) (interface{
...
+func builtinDayName(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
+	v, err := builtinWeekDay(args, ctx)
+	if err != nil {
+		return nil, err
+	}
+	if types.IsNil(v) {
+		return nil, nil
+	}
+	weekday := v.(int64)
+	if !((weekday >= 0) && (weekday < int64(len(mysql.WeekdayNames)))) {
```
Comment (coocood) [inline expression/builtin/time.go]: I think it's better to rewrite the expression to 
`weekday < 0 || weekday >= int64(len(mysql.WeekdayNames))`

File: expression/builtin/time.go [hunk 4]
```diff
@@ -162,6 +162,21 @@ func builtinNow(args []interface{}, ctx map[interface{}]interface{}) (interface{
 	return t.RoundFrac(int(fsp))
 }
```
Comment (zimulala) [inline expression/builtin/time.go]: you can add comments like `builtinDayOfMonth`.

File: parser/parser.y (hunk 2 above)
Comment (shenli) [inline parser/parser.y]: Need to add some unit tests in parser/parser_test.go

File: parser/parser.y (hunk 1 above)
Comment (hhkbp2) [inline parser/parser.y]: Fixed.

File: parser/parser.y (hunk 2 above)
Comment (hhkbp2) [inline parser/parser.y]: @zimulala @shenli 
Some simple test cases are added.

File: parser/parser_test.go [hunk 5]
```diff
@@ -363,8 +362,6 @@ func (s *testParserSuite) TestBuiltin(c *C) {
 		{`SELECT LOCATE('bar', 'foobarbar');`, true},
 		{`SELECT LOCATE('bar', 'foobarbar', 5);`, true},
 
-		{"select current_date, current_date(), curdate()", true},
```
Comment (zimulala) [inline parser/parser_test.go]: Why do you delete it?

Comment (hhkbp2) [inline parser/parser_test.go]: @zimulala Please note these test cases are reorganized with some newly added ones from line 402 - 426. They are listed in an order from small time units to big time units.

**Comment by shenli** [comment]
LGTM
Well done!

**Comment by zimulala** [comment]
LGTM

**Comment by ngaut** [comment]
Thanks @hhkbp2

File: mysql/time.go [hunk 6]
```diff
@@ -95,6 +95,17 @@ var (
 	MinTimestamp = time.Date(1970, 1, 1, 0, 0, 1, 0, time.UTC)
 	// MaxTimestamp is the maximum for mysql timestamp type.
 	MaxTimestamp = time.Date(2038, 1, 19, 3, 14, 7, 999999, time.UTC)
+
+	// WeekdayNames lists names of weekdays, which are used in builtin time function `dayname`.
+	WeekdayNames = []string{
```
Comment (disksing) [inline mysql/time.go]: I have noticed that package `time` has already defined weekday names.
See https://godoc.org/time#Weekday.String
Maybe no need to define it again.

Comment (ngaut) [inline mysql/time.go]: @hhkbp2

**Pull request merged**